docs/content/en/docs/contributing.md

```diff
@@ -0,0 +1,123 @@
+<!-- mdformat off(yaml frontmatter) -->
+
+______________________________________________________________________
+
+## title: Contributing to HEIR weight: 1
+
+<!-- mdformat on -->
+
+There are several ways to contribute to HEIR, including:
+
+- Discussing high-level designs and questions on HEIR's
+  [discussions page](https://github.com/google/heir/discussions)
+- Improving or expanding HEIR's documentation
+- Contributing to HEIR's [code-base](https://github.com/google/heir)
+- Discuss project direction at HEIR's
+  [Working Group meetings](https://google.github.io/heir/community/)
+
+## Ways to contribute
+
+We welcome pull requests. Please take a look at issues marked with
+[good first issue](https://github.com/google/heir/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
+or
+[contributions welcome](https://github.com/google/heir/issues?q=is%3Aissue+is%3Aopen+label%3A%22contributions+welcome%22).
+
+For new proposals, please open a GitHub
+[issue](https://github.com/google/heir/issues) or start a
+[discussion](https://github.com/google/heir/discussions) for feedback.
+
+## Contributing to code using pull requests
+
+### Preparing a pull request
+
+Pre-requisites:
+
+1. Sign the
+   [Contributor License Agreement](https://cla.developers.google.com/about)
+   (CLA). See more
+   [here](https://github.com/google/heir/blob/main/CONTRIBUTING.md#sign-our-contributor-license-agreement).
+
+1. Fork the HEIR repository by clicking the **Fork** button on the
+   [repository page](https://github.com/google/heir). This creates a copy of the
+   HEIR repository on your own GitHub account.
+
+1. See [Getting Started](https://google.github.io/heir/docs/getting_started/) to
+   install developer dependencies to build and run tests.
+
+1. Add the HEIR repository as an upstream remote, so you can use sync your
```

Comment:
spotted a small typo here; otherwise, things look good.

It might be nice to explain how the process differs from the usual GitHub flow - I'd guess other Google projects using similar tooling needed this, too, and might already have documentation/ a blog post we could link to?